docroot/sites/all/modules/contrib/user_cancel_immediate/user_cancel_immediate.module

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * @file
+ * Module file for user_cancel_immediate
+ * Change user cancellation to be immmediate, without any email confirmation.
+ *
+ * Thanks to Clive, http://drupal.stackexchange.com/a/32427/449
+ */
+
+
+/**
+ * Implements hook_form_FORMID_alter()
+ * Adds a custom submit handler
+ */
+function user_cancel_immediate_form_user_cancel_confirm_form_alter(&$form, &$form_state, $form_id) {
+  $form['#submit'][] = '_user_cancel_immediate_user_cancel_form_submit';
```

Comment:
Don't see why we can't just alter the 'user_cancel_confirm' form element to either permanently disable the confirmation email option or to allow a choice for all users:

$form['user_cancel_confirm'] = array(
    '#type' => 'checkbox',
    '#title' => t('Require e-mail confirmation to cancel account.'),
    '#default_value' => FALSE,
    '#access' => FALSE,
    '#description' => t('When enabled, the user must confirm the account cancellation via e-mail.'),
  );